cmd/lib/ci_matrix.rb

```diff
@@ -11,6 +11,7 @@ module CiMatrix
     { symbol: :big_sur,  name: "macos-11", arch: :intel } => 0.0,
     { symbol: :monterey, name: "macos-12", arch: :intel } => 0.0,
     { symbol: :ventura,  name: "macos-13", arch: :intel } => 1.0,
+    { symbol: :sonoma, name: "macos-14", arch: :intel }   => 0.0,
```

Comment:
```suggestion
    { symbol: :sonoma, name: "macos-14", arch: :arm }   => 0.0,
```

It's actually an arm64 runner: https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

Reply:
https://github.com/Homebrew/homebrew-cask/pull/168076